src/llama.cpp

```diff
@@ -5442,6 +5472,9 @@ static void llm_load_vocab(
             } else if (
                 tokenizer_pre == "codeshell") {
                 vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_CODESHELL;
+            } else if (
+                tokenizer_pre == "exaone") {
+                vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_EXAONE;
```

Comment:
This also needs to be handled in `src/llama-vocab.cpp`. I think the pre-tokenizer is similar to starcoder, and so this should be added somewhere over there:

```diff
diff --git a/src/llama-vocab.cpp b/src/llama-vocab.cpp
index 11fffce9..17deefaa 100644
--- a/src/llama-vocab.cpp
+++ b/src/llama-vocab.cpp
@@ -388,6 +388,7 @@ struct llm_tokenizer_bpe {
             case LLAMA_VOCAB_PRE_TYPE_COMMAND_R:
             case LLAMA_VOCAB_PRE_TYPE_SMOLLM:
             case LLAMA_VOCAB_PRE_TYPE_CODESHELL:
+            case LLAMA_VOCAB_PRE_TYPE_EXAONE:
                 regex_exprs = {
                     "\\p{N}",
                     "'s|'t|'re|'ve|'m|'ll|'d| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)",
```